invenio_records_resources/records/api.py

```diff
@@ -131,42 +132,42 @@ def __init__(self, object_model=None, file_model=None):
         self.file_model = file_model or object_model.file
 
     @classmethod
-    def from_dict(cls, data, bucket):
+    def from_dump(cls, data):
         """Construct a file wrapper from a dictionary."""
         file_args = dict(
             id=data["file_id"],
-            storage_class=data.get("storage_class"),
             size=data.get("size"),
             checksum=data.get("checksum"),
         )
-        if "uri" in data:
-            file_args["uri"] = data["uri"]
-
         fi = FileInstance(**file_args)
         obj = ObjectVersion(
-            version_id=data["version_id"],
+            version_id=data["object_version_id"],
             key=data["key"],
             file_id=data["file_id"],
+            file=fi,
             _mimetype=data["mimetype"],
             is_head=True,
-            bucket=bucket,
-            bucket_id=data.get("bucket_id", bucket.id),
+            bucket_id=data["bucket_id"],
         )
         return cls(object_model=obj, file_model=fi)
 
     def dumps(self):
         """Dump file model attributes of the object."""
-        return {
-            "version_id": str(self.object_model.version_id),
-            "key": self.object_model.key,
-            "bucket_id": str(self.object_model.bucket_id),
-            "file_id": str(self.object_model.file_id),
-            "uri": str(self.object_model.file.uri),
-            "storage_class": self.object_model.file.storage_class,
```

Comment:
This might break the check that is currently done by the clients to verify that `FETCH` files have been downloaded.